### ISSUE TITLE
Add support for `input_scaling`

### ISSUE THREAD

**Issue by juanaurre99** [Cyber-Physical-Systems-in-Mech-Eng/pyReCo#69]
Add `input_scaling` as a configurable hyperparameter for the reservoir. This controls the strength of the input signal and plays a key role in balancing nonlinearity and memory.

**Comment by juanaurre99** [comment]
@mertstend I am working in this issue now

**Comment by mertstend** [comment]
Great, thanks @juanaurre99 !! 

can you please display here how you normalize? We should be on the same page before you go ahead implementing. I think that every input channel (last dimension of input data array) should be normalized to -1 to 1 based on the training set samples. But we could also do [0, 1], just in case someone is using sigmoid activation. 

I think that this will make the hyperparameter search for the leakage rate much easier. As of now, the effect of external inputs and their absolute value range interfers with the leakage rate.

**Comment by juanaurre99** [comment]
I was thinking of implementing two helper functions: one for the data normalizations and one for input weight scaling, both according to the activation function. So for tanh [-1,1] and for sigmoid [0,1]

**Comment by mertstend** [comment]
as I think about it: make it [-1, 1] irrespective of the activation. If the user changes the activation later on, we would have to trigger some changes to the normalization as well, I'm afraid that it will over-complicate things. sigmoid(-1) is far from zero, so all good.

**Comment by juanaurre99** [comment]
ok thanks for the tip, I will stick with [-1,1]
